src/decoder/ifd.rs

```diff
@@ -166,14 +201,51 @@ impl Entry {
             // TODO check if this could give wrong results
             // at a different endianess of file/computer.
             (Type::BYTE, 1) => Ok(Unsigned(u32::from(self.offset[0]))),
+            (Type::BYTE, 2) => offset_to_bytes(2, self),
+            (Type::BYTE, 3) => offset_to_bytes(3, self),
+            (Type::BYTE, 4) => offset_to_bytes(4, self),
+            (Type::BYTE, n) => {
+                if n as usize > limits.decoding_buffer_size / mem::size_of::<Value>() {
+                    return Err(TiffError::LimitsExceeded);
+                }
+                let mut v = Vec::with_capacity(n as usize);
```

Comment:
Now would be a good opportunity to factor out these 4 lines, reused across many entry types, into a dedicated method.

Reply:
Done via closure